janusgraph-core/src/main/java/org/janusgraph/diskstorage/locking/consistentkey/ConsistentKeyLocker.java

```diff
@@ -382,13 +382,30 @@ private WriteResult tryWriteLockOnce(StaticBuffer key, StaticBuffer del, StoreTr
         final Timer writeTimer = times.getTimer().start();
         StaticBuffer newLockCol = serializer.toLockCol(writeTimer.getStartTime(), rid, times);
         Entry newLockEntry = StaticArrayEntry.of(newLockCol, zeroBuf);
+        StoreTransaction newTx = null;
         try {
-            final StoreTransaction newTx = overrideTimestamp(txh, writeTimer.getStartTime());
+            newTx = overrideTimestamp(txh, writeTimer.getStartTime());
+
             store.mutate(key, Collections.singletonList(newLockEntry),
                 null == del ? KeyColumnValueStore.NO_DELETIONS : Collections.singletonList(del), newTx);
+
+            newTx.commit();
+            newTx = null;
         } catch (BackendException e) {
             log.debug("Lock write attempt failed with exception", e);
             t = e;
+        } finally {
+            if (newTx != null) {
+                try {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Transaction is still open! Rolling back", new Throwable());
+                    }
+
+                    newTx.rollback();
+                } catch (Throwable excp) {
+                    log.debug("Transaction rollback failed", excp);
```

Comment:
In this situation the transaction will leak. I think we should log here `error` with the explanation that the above transaction is leaked. Of course it is better to somehow track it and let the logic somehow handle the transaction but I guess it makes the implementation slightly harder, so I'm totally fine to just log an error here and add a `TODO` message to actually resolve it later.
Something like:
```
//TODO: We need to handle the transaction newTx which potentially may leak into memory
log.error("Failed to rollback the transaction "+newTx.toString()+". The transaction now may be leaked into memory.", e);
```

Reply:
I agree. Will update the patch shortly. Thanks!